### ISSUE TITLE
IOError: invalid data stream

### ISSUE THREAD

**Issue by kootenpv** [google/categorybuilder#1]
Followed the instructions, but I'm getting errors:

    IOError: invalid data stream # python 2.7

I tried `2to3 -w .` and that seemed to work for converting to python 3 compatible code. Similar error:

    OSError: Invalid data stream # python 3.7

Using arch linux

Does not matter whether I run `category_builder.py` or `analogy.py`. See trace:

```python
pascal@archbook:~/gits/categorybuilder$ python2 category_builder.py ford nixon
Traceback (most recent call last):
  File "category_builder.py", line 29, in <module>
    CB = util.CategoryBuilder()
  File "/home/pascal/gits/categorybuilder/category_builder_util.py", line 116, in __init__
    CreateShelves()
  File "/home/pascal/gits/categorybuilder/category_builder_util.py", line 62, in CreateShelves
    CreateShelf(GetPath(F_TO_I_INPUT), GetPath(F_TO_I_SHELF))
  File "/home/pascal/gits/categorybuilder/category_builder_util.py", line 48, in CreateShelf
    for line in csvreader:
IOError: invalid data stream
```

**Comment by asclines** [comment]
Have you installed [git lfs](https://git-lfs.github.com/) and run `git lfs pull`?

**Comment by amahabal** [comment]
Thanks for the report, kootenpv.

As Alex says, it is likely an LFS issue and you have pointers to the files rather than the actual data. The bz2 files should be over a GB each in size. With an LFS issue, they will be just a few KB.

I have now added diagnostics so that this will be checked by the program, leading to fewer surprises. I also added an initialize.py that is more verbose.

--Abhijit

**Comment by kootenpv** [comment]
It showed to me it was downloading them correctly, but indeed the size is only 1.6G

Thanks for adding the error line. Fixed string formatting in https://github.com/google/categorybuilder/pull/2

I'll give it another shot when I'm not "out".

EDIT: I tried again, and `du -h` shows again 1.6G. Probably that is just my system.

**Comment by amahabal** [comment]
Not sure if you have gotten this to work yet. Can you please try the following? Start with a clean checkout and just try python (instead of python3)?

Thanks,
  amahabal

**Comment by kootenpv** [comment]
I had another attempt with a good network, and just tried it once more with a new pull.

```bash
pascal@archbook:~/gits/categorybuilder$ python2 analogy.py test example
The file ./i-to-f-shelf seems too small.
Did you run 'git lfs pull'? Git stores large files differently.
pascal@archbook:~/gits/categorybuilder$ git lfs pull
Skipping object checkout, Git LFS is not installed./s                                    
```
I noticed that it downloaded the first file (showing 50%, 1/2), it continues very shortly starting with file 2, and then it tells me `Skipping object checkout, Git LFS is not installed./s` which is not true, it downloaded 1.6G.

I tried on a different machine (ubuntu). I noticed that when I did a pull, it also started the download with lfs immediately (that's different behavior on arch linux for me!).

I have gotten it to work on ubuntu, thanks! Something must be wrong with the git-lfs plugin on arch.
